lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -866,6 +896,31 @@ def test_pagination(self):
             }
         )
 
+    @ddt.data(
+        (True, "endorsed_comment"),
+        ("true", "endorsed_comment"),
+        ("1", "endorsed_comment"),
+        (False, "non_endorsed_comment"),
+        ("false", "non_endorsed_comment"),
+        ("0", "non_endorsed_comment"),
+    )
```

Comment:
If this uses the NullBooleanField, we should also have a test that handles the null case, as well an invalid case.  What happens if the value passed is "hello"?  It should raise a validation error. (to address [Brian's concern about silent failures](https://openedx.atlassian.net/browse/MA-1211?focusedCommentId=138440&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-138440)